src/python/bam_to_raw_fastq.py

```diff
@@ -46,39 +46,38 @@ def write_read(fastq, read):
                      'sequence': read.query_sequence})
         
     fastq.write('@{name}\n{sequence}\n+\n{quality}\n'.format(**info))
-    
-def check_putative_barcode(barcode_str, barcode_matching_dict):
+
+def check_putative_barcode(barcode_str, barcode_exact_dict, barcode_mismatch_dict):
```

Comment:
This function is used here and in correct fastq. That means that we need to always make sure that if we change one we will change also the other.  The best thing would be to put them in a util file and source it here.

Reply:
Where should I put a utils file? src/python? Or at the top level?

Reply:
in src/python is fine